pkg/tfgen/docs.go

```diff
@@ -505,7 +502,7 @@ func (p *tfMarkdownParser) parseSection(section []string) error {
 				}
 			}
 			description := strings.Join(subsection, "\n") + "\n"
-			if sectionKind == sectionExampleUsage {
+			if hasExamples {
```

Comment:
Instead of this, what do you think about identifying the example sections in `parse()` and making sure that the titles are nested under an h2? That is, identify all h2's (`##`) that have an example code-block and convert the title of those sections to an h3 (`###`) and insert an h2 called `Example Usage` at the beginning before the first h3 example section.